apps/server/src/shared/repo/dashboard/dashboard.model.mapper.ts

```diff
@@ -42,13 +43,13 @@ export class DashboardModelMapper {
 		return new DashboardEntity(modelEntity.id, { grid, userId: modelEntity.user.id });
 	}
 
-	async mapReferenceToModel(reference: IGridElementReference): Promise<DefaultGridReferenceModel> {
+	mapReferenceToModel(reference: ILearnroom): Course {
 		const metadata = reference.getMetadata();
-		const existingReference = await this.em.findOne(DefaultGridReferenceModel, metadata.id);
-		const result = existingReference || new DefaultGridReferenceModel(metadata.id);
-		result.color = metadata.displayColor;
-		result.title = metadata.title;
-		return result;
+		if (metadata.type === LearnroomTypes.Course) {
```

Comment:
wouldn't be switch cases better for future addings?

Reply:
interesting, I need to think about that one. My initial feeling is that while we only have a single type, the switch would look wierd... let me revisit that on monday